azure-toolkit-libs/azure-toolkit-compute-lib/src/main/java/com/microsoft/azure/toolkit/lib/compute/vm/VirtualMachine.java

```diff
@@ -65,6 +66,36 @@ public Region getRegion() {
         return Region.fromName(remote().regionName());
     }
 
+    public boolean isSshEnabled() {
+        // TODO: @wangmi check if ssh is enabled, possible solution: INBOUND/TCP/22
+        // return remote().getPrimaryNetworkInterface().getNetworkSecurityGroup().securityRules().entrySet().stream()
+        //    .anyMatch(e -> SecurityRuleProtocol.TCP.equals(e.getValue().protocol()) &&
+        //        SecurityRuleDirection.INBOUND.equals(e.getValue().direction()) &&
+        //        "22".equals(e.getValue().destinationPortRange()) &&
+        //        "*".equals(e.getValue().destinationPortRange()));
+        return true;
+    }
+
+    public String getHostIp() {
+        return remote().getPrimaryPublicIPAddress().ipAddress();
```

Comment:
VM may not have a public ip address, we may need to add null check here. Besides, do we also need to add method to create public ip address?